src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -28,7 +28,7 @@
 
 namespace Microsoft.VisualStudio.TestPlatform.Client.Execution;
 
-public class TestRunRequest : ITestRunRequest, ITestRunEventsHandler2
+public class TestRunRequest : ITestRunRequest, IInternalTestRunEventsHandler
```

Comment:
The idea behind changing this is that TestRequestManager does not use ITestRunEventsHandler2 to talk with the outside world, so we can simply have interface that implements the latest version of that public interface, without the old methods, and only adapt the results at the edge, when we give them to an ITestRunEventsHandler2 via event registrar.

Reply:
Additional benefit here is that we don't force AttachDebugger methods on this interface that users use to extend us when we never call them through this interface.